app/controllers/subscriptions_auth_token_controller.rb

```diff
@@ -1,5 +1,25 @@
 class SubscriptionsAuthTokenController < ApplicationController
+  before_action :validate_params
+
   def auth_token
     render json: {}, status: :ok
   end
+
+  def validate_params
+    ParamsValidator.new(expected_params).validate!
+  end
+
+  def expected_params
+    params.permit(:address, :topic_id, :frequency)
+  end
+
+  class ParamsValidator < OpenStruct
+    include ActiveModel::Validations
+
+    validates :address, presence: true
+    validates :address, email_address: true, allow_blank: true
```

Comment:
Could we combine these two lines? I'm wondering if we need `allow_blank` too.

Reply:
👍